src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -699,9 +699,21 @@ private void EmitSequencePointExpression(BoundSequencePointExpression node, bool
         {
             EmitSequencePoint(node);
 
-            // used is true to ensure that something is emitted
-            EmitExpression(node.Expression, used: true);
-            EmitPopIfUnused(used);
+            if (node.Expression != null)
+            {
+                // used is true to ensure that something is emitted
+                EmitExpression(node.Expression, used: true);
+                EmitPopIfUnused(used);
+            }
+            else
```

Comment:
When is `node.Expression == null`?

Reply:
I don't know all the cases that sequence point expressions are used, but this code handled a null expression before, and it is an optional field. I'm trying to make the generated code more robust.

---
In reply to: [184771667](https://github.com/dotnet/roslyn/pull/26138#discussion_r184771667) [](ancestors = 184771667)

Reply:
`EmitExpression(node.Expression, used: true);` does not emit something if `node.Expression` is `null`.

---
In reply to: [184821676](https://github.com/dotnet/roslyn/pull/26138#discussion_r184821676) [](ancestors = 184821676,184771667)

Reply:
`BoundSequencePointExpression.Expression` is marked `Null="disallow"` in BoundNodes.xml. If you feel strongly that we need to handle `node.Expression == null` here, we should return from the method without emitting a sequence point instead.

Reply:
Oh, It was temporarily permitting null while this PR was under development, but I reverted that. I'll revert this change.

Reply:
Not emitting a sequence point for a sequence point node would be wrong.